### PR TITLE
[clangd] Clean formatting modernize-use-override

### LINKED ISSUE #1704: Remove whitespace at the start of the clang-tidy fix modernize-use-override
If i have a class in a header which inherits from a pure virtual base class i observe the following when i apply the fix.

```c++
class A : public Interface
{
    // This will be marked by clangd to use override instead of virtual
    virtual void Reset();

    // After applying the fix i get this and i always have to manually delete the one whitespace :)
     void Reset() override;
};
````
I am not sure if this problem also occurs for other checks where a fix is applied via `clangd`

**System information**
Clangd version (from the log, or `clangd --version`):
```
Apple clangd version 15.0.0 (clang-1500.0.29.1)
Features: mac+xpc
Platform: x86_64-apple-darwin22.5.0; target=arm64-apple-darwin22.5.0
```
clangd extension version: `0.1.24`
Operating system: `macOS Ventura 13.4.1`

### PR DESCRIPTION
**Pull request by kevinjoseph1995**
When applying the recommended fix for the
"modernize-use-override" clang-tidy diagnostic
there was a stray whitespace. This PR fixes that.
Resolves https://github.com/clangd/clangd/issues/1704

### CODE REVIEW

File: clang-tools-extra/clang-tidy/modernize/UseOverrideCheck.cpp [hunk 1]
```diff
@@ -228,8 +228,8 @@ void UseOverrideCheck::check(const MatchFinder::MatchResult &Result) {
   if (HasVirtual) {
     for (Token Tok : Tokens) {
       if (Tok.is(tok::kw_virtual)) {
-        Diag << FixItHint::CreateRemoval(CharSourceRange::getTokenRange(
-            Tok.getLocation(), Tok.getLocation()));
+        Diag << FixItHint::CreateRemoval(CharSourceRange::getCharRange(
+            Tok.getLocation(), Tok.getEndLoc().getLocWithOffset(1)));
```
Comment (PiotrZSL) [inline clang-tools-extra/clang-tidy/modernize/UseOverrideCheck.cpp:232]: Won't work if there are two spaces. You should use LexerUtils::findNextTokenIncludingComments, and then take location -1

File: clang-tools-extra/clangd/unittests/DiagnosticsTests.cpp [hunk 2]
```diff
@@ -898,6 +898,34 @@ TEST(DiagnosticTest, ClangTidySelfContainedDiags) {
                 withFix(equalToFix(ExpectedDFix))))));
 }
 
+TEST(DiagnosticTest, ClangTidySelfContainedDiagsFormatting) {
```
Comment (PiotrZSL) [inline clang-tools-extra/clangd/unittests/DiagnosticsTests.cpp:901]: Tests for clang tidy should be added here: clang-tidy/checkers/modernize/use-override.cpp

**Comment by PiotrZSL** [comment]
Release notes entry (for clang-tidy) is also missing.

File: clang-tools-extra/clangd/unittests/DiagnosticsTests.cpp (hunk 2 above)
Comment (kevinjoseph1995) [reply clang-tools-extra/clangd/unittests/DiagnosticsTests.cpp:901]: Added a case in clang-tools-extra/test/clang-tidy/checkers/modernize/use-override.cpp.

**Comment by kevinjoseph1995** [comment]
> Release notes entry (for clang-tidy) is also missing.

Added a small entry to the Release notes. I wasn't sure about which section to add it to. Please let me know me if that isn't the right one.

File: clang-tools-extra/docs/ReleaseNotes.rst [hunk 3]
```diff
@@ -160,6 +160,10 @@ Changes in existing checks
   `AllowStringArrays` option, enabling the exclusion of array types with deduced
   length initialized from string literals.
 
+- Improved :doc:`
```
Comment (PiotrZSL) [inline clang-tools-extra/docs/ReleaseNotes.rst:163]: fine, but missing check name here

**Review by PiotrZSL — APPROVED** [review]
Small fix in release notes needed.
If with this change tests are passing, then it's looking fine for me.

File: clang-tools-extra/docs/ReleaseNotes.rst (hunk 3 above)
Comment (kevinjoseph1995) [reply clang-tools-extra/docs/ReleaseNotes.rst]: Thanks, added the missing check name.

**Comment by kevinjoseph1995** [comment]
> Small fix in release notes needed. If with this change tests are passing, then it's looking fine for me.

ninja check-clang-tools reports the following:
```
Total Discovered Tests: 2813
Unsupported      :    7 (0.25%)
Passed           : 2804 (99.68%)
Expectedly Failed:    2 (0.07%)
```
And all of the ClangdTests pass as well:
```
[==========] 1232 tests from 179 test suites ran. (86642 ms total)
[  PASSED  ] 1232 tests.
```

**Pull request merged**